panelapp/panelapp_utils.py

```diff
@@ -0,0 +1,160 @@
+import requests
+from django.db import transaction
+from django.utils import timezone
+
+from panelapp.models import PaLocusList, PaLocusListGene
+from seqr.models import LocusList as SeqrLocusList, LocusListGene as SeqrLocusListGene
+from seqr.utils.gene_utils import parse_locus_list_items
+from seqr.utils.logging_utils import SeqrLogger
+from seqr.views.utils.json_to_orm_utils import update_model_from_json, create_model_from_json
+from settings import PANEL_APP_API_URL
+
+logger = SeqrLogger(__name__)
+
+
+def import_all_panels(user):
+    def _extract_ensembl_id_from_json(raw_gene_json):
+        ensembl_genes_json = raw_gene_json.get('gene_data', {}).get('ensembl_genes')
+        if ensembl_genes_json and isinstance(ensembl_genes_json, dict):
+            return ensembl_genes_json \
+                .get('GRch38', {}) \
+                .get('90', {}) \
+                .get('ensembl_id')
+        else:
+            return None
+
+    panels_url = '{}/panels/?page=1'.format(PANEL_APP_API_URL)
+
+    all_panels = _get_all_panels(panels_url, [])
+
+    for panel in all_panels:
+        with transaction.atomic():
```

Comment:
is there any risk of successfully updating some of the panel app lists but then having subsequent ones fail? Like if that happens will that be okay, or will that be bad?

Reply:
That's a good point, have fixed the code to continue with importing remaining panels even if one fails unexpectedly.